### PR TITLE
(#594) Update to Node 16 and npm 8

### LINKED ISSUE #594: Update app to use node 16 and npm 6
### Issue description
Update the `engines` in package.json to the new versions, and update the GH workflow. We will also have to update package-lock.json to use the new npm format.

As part of this ticket, we need to update to the newest release of the `cgov-react-app-playground` shared workflow.

> **ESTIMATE** 2

### Additional details / screenshot
- Screenshot

### Related Tickets
* React App Playground: https://github.com/NCIOCPL/cgov-react-app-playground/issues/74

### LINKED ISSUE #595: Change out node-sass for sass
`node-sass` has been deprecated for almost 2 years now. `sass` is the correct module to use. Additionally, the version of node-sass that this repository uses, and many of the others, does not work well with newer node/npm combinations. (Basically Node 12 and NPM 6 works -- anything outside of that an YMMV).

So... is it just updating? Based on experience with CGDP, NO. Some things to check:
1. `sass-loader` might need to be bumped.
2. Your sass may suck and things that should have failed will fail now. It could also be one of the situations where `sass` does not do something node-sass did and that is the way they are going to keep it.

The following is what we did on the platform and react-app-playground for both bumping Node, Updating loaders, and moving from node-sass to sass. May your experience go smoother -- definitely a lot less SASS to go through, and newer to boot.
* https://github.com/NCIOCPL/cgov-digital-platform/pull/3258
* https://github.com/NCIOCPL/cgov-digital-platform/pull/3332
* https://github.com/NCIOCPL/cgov-react-app-playground/pull/62

### PR DESCRIPTION
**Pull request by adriancofie**
* Closes #594 
* Closes #595

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -224,10 +224,10 @@
     "jest-resolve": "^27.0.6",
     "jest-watch-typeahead": "^0.4.2",
     "nock": "^11.7.0",
-    "node-sass": "^4.14.1",
     "prop-types": "^15.7.2",
     "redux-devtools-extension": "^2.13.8",
     "redux-mock-store": "^1.5.3",
+    "sass": "^1.55.0",
```
Comment (seyilonge-nci) [inline package.json:230]: Any reason why this is being changed in this PR?

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v14.17.3
+16
```
Comment (seyilonge-nci) [inline .nvmrc:1]: Can you replace this with `lts/gallium`

**Review by seyilonge-nci — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (adriancofie) [reply package.json:230]: Req'd for Node 16

Comment (seyilonge-nci) [reply package.json:230]: Thought as much, but believed that was being tackled in a separate PR. Just noticed this PR has two commits. Cool!

**Review by seyilonge-nci — APPROVED** [review]

**Comment by arcepaul** [comment]
Should this also be the point at which we point to `workflow/v2` in workflow.yml?

**Review by arcepaul — APPROVED** [review]

**Pull request merged**